staging/src/k8s.io/cloud-provider/controllers/node/metrics.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"sync"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	metricsSubsystem = "node_controller"
+)
+
+var (
+	removeCloudProviderTaintDelay = metrics.NewHistogram(
+		&metrics.HistogramOpts{
+			Subsystem:      metricsSubsystem,
+			Name:           "remove_cloud_provider_taint_delay_seconds",
+			Help:           "Number of seconds after node creation when NodeController removed the cloud-provider taint of a single node.",
+			Buckets:        metrics.ExponentialBuckets(1, 4, 5), // 1s -> ~4m [1, 4, 16, 64, 256]
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+	initializeNodeAddressDelay = metrics.NewHistogram(
+		&metrics.HistogramOpts{
+			Subsystem:      metricsSubsystem,
+			Name:           "initial_node_addresses_delay_seconds",
+			Help:           "Number of seconds after node creation when NodeController initially updated the address of a single node.",
+			Buckets:        metrics.ExponentialBuckets(1, 4, 5), // 1s -> ~4m [1, 4, 16, 64, 256]
```

Comment:
this goes beyond 4m for very large clusters, can we get the upperbound  to ~ 15 mins?

Reply:
Sure, I'll bump 5 buckets to 6.